models/autoscalers.go

```diff
@@ -0,0 +1,62 @@
+package models
+
+import (
+	"k8s.io/api/autoscaling/v1"
+	"time"
+)
+
+type Autoscalers []Autoscaler
+type Autoscaler struct {
+	Name      string            `json:"name"`
+	Labels    map[string]string `json:"labels"`
+	CreatedAt string            `json:"created_at"`
+	// Spec
+	MinReplicas                    int32 `json:"min_replicas"`
+	MaxReplicas                    int32 `json:"max_replicas"`
+	TargetCPUUtilizationPercentage int32 `json:"target_cpu_utilization_percentage"`
+	// Status
+	ObservedGeneration              int64  `json:"observed_generation,omitempty"`
+	LastScaleTime                   string `json:"last_scale_time,omitempty"`
+	CurrentReplicas                 int32  `json:"current_replicas"`
+	DesiredReplicas                 int32  `json:"desired_replicas"`
+	CurrentCPUUtilizationPercentage int32  `json:"current_CPU_utilization_percentage,omitempty"`
+}
+
+func (autoscalers *Autoscalers) Parse(ds *v1.HorizontalPodAutoscalerList) {
+	if ds == nil {
+		return
+	}
+
+	for _, autoscaler := range ds.Items {
+		casted := Autoscaler{}
+		casted.Parse(&autoscaler)
+		*autoscalers = append(*autoscalers, casted)
+	}
+}
+
+func (autoscaler *Autoscaler) Parse(d *v1.HorizontalPodAutoscaler) {
+	autoscaler.Name = d.Name
+	autoscaler.Labels = d.Labels
+	autoscaler.CreatedAt = d.CreationTimestamp.Time.Format(time.RFC3339)
```

Comment:
This format invocations could hide some panic call, if time.RFC3339 is always not nil with correct data is fine, or if Time.Format() is renders a 0 createdAt also is valid. But worth to check the method to avoid a 50X error due a non expected panic.

Reply:
Same comment for other uses of Format()

Reply:
When there is an error on Format() method it returns a date from 1970-ish. I do trust in other objects in the `d.CreationTimestamp.Time`
* d.CreationTimestamp - it is pretty hard that they forgot to return it.
* d.CreationTimestamp.Time - it is a wrapper of a time.Time type, so it has to have always a date.
* time.RFC3339 - it is a constant within the built-in time library, so it has always a valid value.